prereise/gather/hiflddata/data_trans.py

```
@@ -0,0 +1,645 @@
+    nodes = [node]
+    for i in range(1, depth + 1):
+        output[i] = []
+        for x in nodes:
+            output[i].extend(layers.get(x, []))
+        nodes = output[i]
+    return output
+
+
+def Cal_KV(N_dict, G, KV_dict, to_cal):
+    """Estimate missing substation's KV data using the neighbors average calculation
+
+    :param dict N_dict: dict of substation's neighbors as returned by :func:`Neighbors`
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param dict KV_dict: dict of existing baseKV for each substation
+    :param list to_cal: list of the substations to be calculate using average KV
+    :return: (*dict*) --  dict of all substation's baseKV being corrected and calculated.
+    """
+
+    for sub in to_cal:
+        if sub not in N_dict:
+            continue
+        neigh = get_neigbors(G, sub, depth=5)
+        temp_KV = 0
+        num = 0
+        for i in range(1, 6):
+            for nei in neigh[i]:
+                if nei in KV_dict:
+                    temp_KV = temp_KV + KV_dict[nei]
+                    num = num + 1
+            if num > 0:
+                KV_dict[sub] = temp_KV / num
+                continue
+            else:
+                KV_dict[sub] = -999999
+    return KV_dict
+
+
+def Set_Sub(E_csv):
+    """Generate the subs
+
+    :param str E_csv: path of the HIFLD substation csv file
+    :return: (*dict*) --  sub_by_coord_dict, a dict mapping from (x, y) to substation detail.
+    :return: (*dict*) --  sub_name_dict, a dict mapping from substation name to its coordinate (x, y).
+    """
+
+    sub_by_coord_dict = {}
+    sub_name_dict = {"sub": []}
+    raw_subs = pd.read_csv(E_csv)
+    for index, row in raw_subs.iterrows():
+        location = (row["LATITUDE"], row["LONGITUDE"])
+        if location in sub_by_coord_dict:
+            print("WARNING: substations coordinates conflict check:", location)
+        sub_by_coord_dict[location] = (row["ID"], row["NAME"])
```

It seems dangerous to simply overwrite the data with a printed warning to the user. I think it would be better to raise an Exception.

---

agree, will raise an exception in the next revision.